pkg/client/e2e.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/results"
+	"github.com/pkg/errors"
+)
+
+// GetTests extracts the junit results from a sonobuoy archive and returns the requested tests.
+func (c *SonobuoyClient) GetTests(reader io.Reader, show string) (TestCases, error) {
+	read := results.NewReaderWithVersion(reader, "irrelevant")
+	junitResults := jUnitResults{}
+	err := read.WalkFiles(func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		return results.ExtractFileIntoStruct("plugins/e2e/results/junit_01.xml", path, info, &junitResults)
+	})
+	if err != nil {
+		return TestCases{}, errors.Wrap(err, "failed to walk results archive")
+	}
+	out := make(TestCases, 0)
+	if show == "passed" || show == "all" {
+		out = append(out, junitResults.filter(passed)...)
+	}
+	if show == "failed" || show == "all" {
+		out = append(out, junitResults.filter(failed)...)
+	}
+	if show == "skipped" || show == "all" {
+		out = append(out, junitResults.filter(skipped)...)
+	}
+	sort.Sort(out)
+	return out, nil
+}
+
+type TestCases []TestCase
+
+func (t TestCases) Len() int           { return len(t) }
+func (t TestCases) Less(i, j int) bool { return t[i].Name < t[j].Name }
+func (t TestCases) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+
+func (t TestCases) String() string {
+	var buf bytes.Buffer
+	for _, tc := range t {
+		fmt.Fprintln(&buf, tc.Name)
+	}
+	return buf.String()
+}
+
+// Focus returns a value to be used in the E2E_FOCUS variable that is
+// representative of the test cases in the struct.
+func (t TestCases) Focus() string {
+	// YAML doesn't like escaped characters and regex needs escaped characters. Therefore a double escape is necessary.
+	r := strings.NewReplacer("[", `\\[`, "]", `\\]`)
+	testNames := make([]string, len(t))
+	for i, tc := range t {
+		testNames[i] = r.Replace(tc.Name)
+	}
+	return strings.Join(testNames, "|")
+}
+
+// TestCase is a single junit test result.
+type TestCase struct {
```

Comment:
These are defined in ginko as well, I could pull in their structs but didn't want to include ginko as a dep.

Reply:
https://github.com/onsi/ginkgo/blob/master/reporters/junit_reporter.go seems pretty bounded, I'd move this code into the tarball utils and yank it in.